### PR TITLE
feat: update extension for qupath v0.2.0 m10

### PR DESCRIPTION
**Pull request by arvgord**
- Updated extension for QuPath 0.2.0-m10;
- Updated JFoenix dependency to latest;
- Updated Readme.md
- Changed JFXTextField to standard JavaFX TextField to prevent the following error: https://stackoverflow.com/questions/55889654/illegalaccessexception-for-jfxtextfield-with-java-sdk-12
- Resolved URISyntaxException in Windows

### CODE REVIEW

File: src/main/java/com/quantumsoft/qupathcloud/gui/qupath/CloudExtension.java [hunk 1]
```diff
@@ -63,10 +66,11 @@ public void installExtension(QuPathGUI qupath) {
     });
     logoutButton.disableProperty().bind(Repository.INSTANCE.getIsLoggedInProperty());
 
-    qupath.addToolbarSeparator();
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/gui/qupath/CloudExtension.java:66]: The tool bar separator seems to be removed - is this ok?

**Review by poojavenkatram — COMMENTED** [review]
Hi Artyom,

Thanks for the changes - looks good for the most part. Just one quick check and will merge and release after this is in.

Best
Pooja

File: src/main/java/com/quantumsoft/qupathcloud/gui/qupath/CloudExtension.java (hunk 1 above)
Comment (arvgord) [reply src/main/java/com/quantumsoft/qupathcloud/gui/qupath/CloudExtension.java:66]: Thank you! I missed that. Added.

Comment (poojavenkatram) [reply src/main/java/com/quantumsoft/qupathcloud/gui/qupath/CloudExtension.java:66]: Thanks :)

File: src/main/java/com/quantumsoft/qupathcloud/repository/Repository.java [hunk 2]
```diff
@@ -106,10 +108,7 @@
     cloudDao.set(new CloudDaoImpl(oAuth20));
 
     hierarchyListener = new QuPathHierarchyListener();
-    qupath.addImageDataChangeListener((source, imageDataOld, imageDataNew) -> {
-      LOGGER.trace("ImageData change old: " + imageDataOld + ", new:" + imageDataNew);
-      hierarchyListener.setListenedImageData(imageDataNew);
```
Comment (poojavenkatram) [inline src/main/java/com/quantumsoft/qupathcloud/repository/Repository.java:111]: we are not setting this ImageListenedData in the hierarchy listener

**Review by poojavenkatram — COMMENTED** [review]
Thanks Artyom - just one more question.

Best
Pooja

File: src/main/java/com/quantumsoft/qupathcloud/repository/Repository.java (hunk 2 above)
Comment (arvgord) [reply src/main/java/com/quantumsoft/qupathcloud/repository/Repository.java:111]: I moved this code to the following part:
```
    @Override
    public void changed(ObservableValue<? extends ImageData<BufferedImage>> source,
        ImageData<BufferedImage> imageDataOld, ImageData<BufferedImage> imageDataNew) {
      LOGGER.trace("ImageData change old: " + imageDataOld + ", new:" + imageDataNew);
      if (this.imageData != null) {
        this.imageData.getHierarchy().removePathObjectListener(this);
      }
      if (imageDataNew != null) {
        this.imageData = imageDataNew;
        this.imageData.getHierarchy().addPathObjectListener(this);
      }
    }
```

Comment (poojavenkatram) [reply src/main/java/com/quantumsoft/qupathcloud/repository/Repository.java:111]: LGTM

**Comment by poojavenkatram** [comment]
Thanks for the changes Artyom - LGTM :)

**Comment by arvgord** [comment]
Thank you! Could you release a new version of the extension to close [this](https://github.com/GoogleCloudPlatform/qupath-chcapi-extension/issues/35) issue, please?

**Pull request merged**